eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -983,13 +983,15 @@ protected function copySubtree($actionMap, $oldParentAliasId, $newParentAliasId,
 
             $alreadyGeneratedAliases[] = $identifier;
 
-            $alreadyGeneratedAliases = array_merge(
-                $alreadyGeneratedAliases,
-                $this->copySubtree(
-                    $actionMap,
-                    $oldParentAliasId,
-                    $row['id'],
-                    $alreadyGeneratedAliases
+            $alreadyGeneratedAliases = array_unique(
```

Comment:
Is it possible to pass `$alreadyGeneratedAliases` as an reference instead of doing array_unique(array_merge(...)) ?

Reply:
I have already passed `$alreadyGeneratedAliases` as reference but that was requested to change in the review: https://github.com/ezsystems/ezpublish-kernel/pull/3127#discussion_r741882342

Reply:
@alongosz Could you please explain why we shouldn't use an reference here?

Reply:
As far as I know it's only a general practice. I noted it the first time it appeared and was ok with the change (to normal variable) at the time, but I can see clear benefits.

In this case, it actually makes sense, because it would make it unnecessary to perform the `array_merge`, so there is a substantial performance gain to be had (no numbers tho, but you can imagine the complexity of the merger).

Reply:
> @alongosz Could you please explain why we shouldn't use an reference here?

It's general anti-pattern because it's error-prone, especially when combined with recursive calls. If you need to modify a state, you should use an object (maybe some sort of a collection?). Otherwise a new instance should be returned.

Reply:
Actually, it does not need to be a reference.

If the `$this->copySubtree` returns the original array with new entries, then there is no need to merge it. To be frank, it should have not be merged in the first place :sweat_smile: